tests/phpunit/tests/hooks/HooksTrait.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace phpunit\tests\hooks;
+
+use MockAction;
+use WP_Hook;
+
+trait HooksTrait {
+
+	public function data_not_valid_callback() {
+		$obj = new MockAction();
+
+		return array(
+			'function callback'                 => array(
+				'callback'           => 'not_callable_callback',
+				'callback_as_string' => 'not_callable_callback',
+			),
+			'static method callback as string'  => array(
+				'callback'           => 'Foo::not_callable_callback',
+				'callback_as_string' => 'Foo::not_callable_callback',
+			),
+			'static method callback as array'   => array(
+				'callback'           => array( 'Foo', 'not_callable_callback' ),
+				'callback_as_string' => 'Foo::not_callable_callback',
+			),
+			'object method callback'            => array(
+				'callback'           => array( $obj, 'not_callable_callback' ),
+				'callback_as_string' => 'MockAction::not_callable_callback',
+			),
+			'a boolean type is not a callback'  => array(
+				'callback'           => false,
+				'callback_as_string' => 'boolean',
+			),
+			'an integer type is not a callback' => array(
+				'callback'           => 100,
+				'callback_as_string' => 'integer',
+			),
```

Comment:
I'd like to suggest an empty array as a callback too (this will trigger a warning in `convert_callback_to_string`):

```suggestion
			),
			'an empty array' => array(
				'callback'           => array(),
				'callback_as_string' => 'integer',
			),
```